### PR TITLE
[WebAssembly] Implement prototype f16x8.splat instruction.

### PR DESCRIPTION
**Pull request by brendandahl**
Adds a builtin and intrinsic for the f16x8.splat instruction.

Specified at:
https://github.com/WebAssembly/half-precision/blob/29a9b9462c9285d4ccc1a5dc39214ddfd1892658/proposals/half-precision/Overview.md

Note: the current spec has f16x8.splat as opcode 0x123, but this is incorrect and will be changed to 0x120 soon.

### CODE REVIEW

**Comment by brendandahl** [comment]
cc @aheejin @dschuff 

As mentioned in the meeting, it looks like it will be a lot more work to get half value's working with normal patterns, so for now I'll stick to just built-ins and intrinsics.

File: clang/lib/Basic/Targets/WebAssembly.h [hunk 1]
```diff
@@ -90,6 +90,7 @@ class LLVM_LIBRARY_VISIBILITY WebAssemblyTargetInfo : public TargetInfo {
 
   StringRef getABI() const override;
   bool setABI(const std::string &Name) override;
+  bool useFP16ConversionIntrinsics() const override { return false; }
```
Comment (dschuff) [inline clang/lib/Basic/Targets/WebAssembly.h:93]: Is this used?

Comment (brendandahl) [reply clang/lib/Basic/Targets/WebAssembly.h:93]: Yeah, this is what causes clang to start outputting `half` types. I could conditionally enable this with `return !HasHalfPrecision;` instead. Though doing a quick test with scalar `__fp16` in c, the `half` types seems to work correctly regardless of this setting.

Comment (aheejin) [reply clang/lib/Basic/Targets/WebAssembly.h:93]: I am not very familiar with the places where the function is used, but `!HasHalfPrecision` seems safer.

**Review by aheejin — APPROVED** [review]
LGTM % `!HasHalfPrecision` thing

By the way I guess you can try getting commit access soon? I think it is still "Send an email to Chris" though...

**Comment by brendandahl** [comment]
> LGTM % `!HasHalfPrecision` thing
> 
> By the way I guess you can try getting commit access soon? I think it is still "Send an email to Chris" though...

Done, can I get a squash and merge?

I'll look into getting commit access.

**Comment by aheejin** [comment]
I guess you've likely already found it but anyway it's here: https://llvm.org/docs/DeveloperPolicy.html#obtaining-commit-access
It's basically "Email Chris".

**Pull request merged**